src/pyFAI/gui/pilx/MainWindow.py

```diff
@@ -118,7 +119,17 @@ def initData(self,
                  ):
 
         self._file_name = os.path.abspath(file_name)
-        self._dataset_path = dataset_path
+
+        c=0
+        for i in dataset_path[-1::-1]:
+            if i.isdigit():
+                c+=1
+            else:
+                break
```

Comment:
I guess we can assume more on the dataset path? 

Do something like `i = dataset_path.find("images_"); _dataset_path = dataset_path[i+6:]; `

Reply:
I would prefer assume as little as reasonably possible. Why enforce the basename if not needed ?

Reply:
I mean, we are still enforcing a particular format of basename here (it must have digits at the end and be contained in a group). So I thought that putting stronger conditions would allow the parsing to be more explicit.

Reply:
it would suppose changing the initData attributes again, but if we pass `group_path` and `dataset_name` on the initData method, with `dataset_name` None as default, and in this case, just iterates a get every dataset inside the group. That would be a general solution

Reply:
Loic, it just imposes all datasets have the same prefix. If not, the `sort`operation will not work as expected.
It is really a pity there is no way to enforce the order of dataset/groups in an HDF5 file. Those are just work around for this weakness. 

Edgar, I am not in favour of changing the signature of methods which are supposed to be in the public API, even though I have no clue if anybody else is using it. As Linus said in https://lkml.org/lkml/2012/12/23/75 ... don't break users' programs. I like Linus because I feel really moderated in comparison with him :)